### ISSUE TITLE
DataTable Checkbox filter fails with multiple checkbox fieldsets

### ISSUE THREAD

**Issue by michael-milette** [wet-boew/wet-boew#8558]
Hi @thekodester ,

We did a little more extensive testing today and I believe we found a bug.

## Description

I have a datatable with two filter fields: Pillars and Provinces. Checking a box in the first field and none in the second field produces 0 results.

## Reproducing the issue

Create a form similar to the working example but with two filter fields of check boxes. The first will be Pillars and the second Provinces in this example.

For your convenience, here are links to two sample pages. Other than the title, the only difference between these two pages is the order in which the filter fields appear.

* https://cfc-swc.gc.ca/test/datatables1.html
* https://cfc-swc.gc.ca/test/datatables2.html

### Testing

* Test 1: Checking a box ONLY in the first filter field in the first field set - **Fail. Produces 0 results.**
* Test 2: Checking boxes in both filter fields - Success. Produces the expected results.
* Test 3: Checking a box ONLY in the second filter field - Success. Produces the expected results.

Now re-do the test using the second page. Notice that Test 1 still fails and Test 2 and 3 still pass even though they are not the same fields.

#### Test 1
![image](https://user-images.githubusercontent.com/3808579/52495165-e4ccd600-2b9d-11e9-9bd6-a4884bd35598.png)

#### Test 2
![image](https://user-images.githubusercontent.com/3808579/52495208-0037e100-2b9e-11e9-8301-6733ad4f9ca6.png)

#### Test 3
![image](https://user-images.githubusercontent.com/3808579/52495253-1ba2ec00-2b9e-11e9-887d-905ead834c28.png)

#### Test 1
![image](https://user-images.githubusercontent.com/3808579/52494373-cebe1600-2b9b-11e9-990d-1e7c6fae1cb1.png)
#### Test 2
![image](https://user-images.githubusercontent.com/3808579/52494447-0462ff00-2b9c-11e9-8e1d-57d9965d5ca3.png)
#### Test 3
![image](https://user-images.githubusercontent.com/3808579/52494480-22306400-2b9c-11e9-828f-679c77a72f62.png)

## What was expected

Regardless of the order in which the filter fields appear, behavior should be consistent between the first and second field. Checking one or more of the checkbox in the first filter fieldset without any in the second should not always result in 0 results.

For example, checking just "Alberta" should produce the same results regardless of the order in which the search fields appear it should not be an empty data set.

Let me know if you have any questions.

Best regards,

Michael

**Comment by thekodester** [comment]
This is expected behaviour and how it works with even the drop-downs. Cause it looking for "GBV Strategy Pillar: Prevention" the whole word and not just "Prevention".

**Comment by thekodester** [comment]
Not it not that either. It has nothing to do with multiple checkboxes, cause do a quick test with a alert or console log it passing the string.

**Comment by thekodester** [comment]
Found the issue has too do with:
```
if ( $value ) {
			$value = $value.replace( /\s/g, "\\s*" );
			$datatable.column( parseInt( $elm.attr( "data-column" ), 10 ) ).search( "(" + $value + ")", true ).draw();
		}
```
Basically the value is never null or empty, so it continues to search incorrect columns. Working on a better way to determine if we change columns.

**Comment by thekodester** [comment]
Posted a fix for it that was erase the value when it starts checking the next set of checkboxes.

**Comment by michael-milette** [comment]
Hi @thekodester ,

Thank you so much for the quick fix. So much better. You the man!!! :-)

Best regards,

Michael

**Comment by michael-milette** [comment]
Hi @duboisp , looking forward to this being merged.

Best regards,

Michael

**Comment by duboisp** [comment]
@michael-milette FYI it is now merged and will be included in the release v4.0.30

**Comment by michael-milette** [comment]
Thank you @dubois! Much appreciated.

Brst regards,

Michael Milette
